okhttp/src/main/java/okhttp3/internal/http2/Http2Codec.java

```diff
@@ -124,6 +124,9 @@ public Http2Codec(OkHttpClient client, Interceptor.Chain chain, StreamAllocation
   @Override public Response.Builder readResponseHeaders(boolean expectContinue) throws IOException {
     List<Header> headers = stream.takeResponseHeaders();
     Response.Builder responseBuilder = readHttp2HeadersList(headers);
+    if (client.protocols().contains(Protocol.H2C)) {
```

Comment:
why set it here and not inside readHttp2HeadersList?

Reply:
That method was `public static` and so I would need to thread the value through the method signature, causing a breaking API change.

Reply:
This could actually be cached on the class. No need to do the linear scan all the time.

Reply:
Alternatively, another attribute could be added to the OkHttpClient and that can be a cached referenced for the lifetime of the client